### PR TITLE
Submit a proposal

### PR DESCRIPTION
**Pull request by AllanEngland**
This branch adds a proposal submission panel to the GUI.

It's highlighted some things that need to be looked at now, I'll add some discussion [here](https://github.com/orgs/orppst/discussions/3) and will also link in any associated API updates.

### CODE REVIEW

File: src/main/webui/src/proposal/Submit.tsx [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    });
+
+    return (
+        <Box>
+            <Text fz="lg" fw={700}>Submit Proposal</Text>
+
+            <Box m={20}>This is where we can have a validation overview, the following are <b>not</b> real!
+                <Table>
+                    <Table.Tbody>
+                        <Table.Tr><Table.Td>✅</Table.Td><Table.Td>Summary</Table.Td></Table.Tr>
```
Comment (alan-stokes) [inline src/main/webui/src/proposal/Submit.tsx:100]: this section to be honest, you likely want it as a separate component itself. That being coz your going to need to call the separate pages validation logic somehow (i think on pages with just forms, you could just separate out the method that makes the form, and call that, and then call validate on the form. check for errors, and if there are any, feed them into your table). For pages with other types of validation (table validation for example) i think it'll need a bit of thinking. 

Another question for you on this, if there are errors, do you plan to route them to the fault? or embed the page on this page?

File: src/main/webui/src/proposal/Submit.tsx [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+                        <Table.Tr><Table.Td>✅</Table.Td><Table.Td>Investigators</Table.Td></Table.Tr>
+                        <Table.Tr><Table.Td>✅</Table.Td><Table.Td>Targets</Table.Td></Table.Tr>
```
Comment (alan-stokes) [inline src/main/webui/src/proposal/Submit.tsx:102]: as a wee optimisation, the tick (and cross when you do it) probably need to use the IconCheck and IconX from the  '@tabler/icons-react'; library, which will make it consistent with all our other icons.

File: src/main/webui/src/proposal/Submit.tsx (hunk 1 above)
Comment (AllanEngland) [reply src/main/webui/src/proposal/Submit.tsx]: I've moved this in a separate component.  I don't know which is preferable out of routing to faults or embedding them on this page.  Hopefully when we come it there'll be enough time to play around with both methods and choose the best one.

File: src/main/webui/src/proposal/Submit.tsx (hunk 2 above)
Comment (AllanEngland) [reply src/main/webui/src/proposal/Submit.tsx]: Good point, I've updated them to React icons.

**Comment by AllanEngland** [comment]
Dependent pull requests have been raised for [pst-lib](https://github.com/orppst/pst-lib/pull/3) and [pst-api-service](https://github.com/orppst/pst-api-service/pull/20)

**Comment by AllanEngland** [comment]
I'm merging this in now as it's working and I'd like to get the changes in before fixing [this](https://github.com/orppst/pst-gui/issues/91).  The blocker is a problem that's effectively hidden now as submitted proposals are not visible.

**Pull request merged**